File: top_100_liked_algos_and_data_structures/sort_binary_array_linear_time/sort.c
#include <stdio.h>

void sort(int arr[], int n) {
    int k = 0;
    for (int i=0; i<n; i++) {
        if (arr[i] == 0) {
            arr[k] = 0;
            k++;
        }
    }

    for (int i=k; i<n; i++) {
        arr[i] = 1;
    }

    for (int i=0; i < n; i++) {
        printf(" %d ", arr[i]);
    }
    printf("\n");
}

void main() {
    int arr[] = {1, 0, 1, 1,1, 0, 0, 0, 0, 1, 1, 0, 1, 0};
    int n = sizeof(arr) / sizeof(arr[0]);
    sort(arr, n);
}

File: top_100_liked_algos_and_data_structures/sort_binary_array_linear_time/sort.py
def sort(arr):
    k = 0
    for i in range(len(arr)):
        if arr[i] == 0:
            arr[k] = 0
            k+=1
        
    for i in range(k, len(arr)):
        arr[i] = 1

    return arr

print(sort([1, 0, 1, 1,1, 0, 0, 0, 0, 1, 1, 0, 1, 0]))